### PR TITLE
owner must be a billing contact, cannot be removed

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner @leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 1]
```diff
@@ -188,6 +188,15 @@ class AdminPaymentsController @Inject() (
     Ok(planCommander.unfreeze(orgId).toString)
   }
 
+  def addOrgOwnersAsBillingContacts() = AdminUserAction { implicit request =>
+    db.readWrite { implicit session =>
+      organizationRepo.allActive.map { org =>
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala]: `foreach`

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 2]
```diff
@@ -188,6 +188,15 @@ class AdminPaymentsController @Inject() (
     Ok(planCommander.unfreeze(orgId).toString)
   }
 
+  def addOrgOwnersAsBillingContacts() = AdminUserAction { implicit request =>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:191]: Nice, thanks!

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 3]
```diff
@@ -188,6 +188,15 @@ class AdminPaymentsController @Inject() (
...
+    db.readWrite { implicit session =>
+      organizationRepo.allActive.map { org =>
+        planCommander.addUserAccountContactHelper(org.id.get, org.ownerId, ActionAttribution(user = None, admin = request.adminUserId))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:194]: Does this method complain if the contact being added is already there?

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:194]: no

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala [hunk 4]
```diff
@@ -313,7 +325,7 @@ class PlanManagementCommanderImpl @Inject() (
   def addUserAccountContactHelper(orgId: Id[Organization], userId: Id[User], attribution: ActionAttribution)(implicit session: RWSession): Option[AccountEvent] = {
     val account = paidAccountRepo.getByOrgId(orgId)
     if (!account.userContacts.contains(userId)) {
-      val updatedAccount = account.copy(userContacts = account.userContacts.filter(_ != userId) :+ userId)
+      val updatedAccount = account.copy(userContacts = account.userContacts :+ userId)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala:313]: good catch, i wonder why that filter was ever in there

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala [hunk 5]
```diff
@@ -104,7 +104,7 @@ class PlanManagementCommanderImpl @Inject() (
   }
 
   //very explicitly accepts a db session to allow account creation on org creation within the same db session
-  private def remainingBillingCycleCost(account: PaidAccount)(implicit session: RSession): DollarAmount = {
+  def remainingBillingCycleCost(account: PaidAccount)(implicit session: RSession): DollarAmount = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Why removed the private?

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: I was going to re-use this for tests, decided not to. Will re-private.

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala [hunk 6]
```diff
@@ -232,13 +232,24 @@ class PlanManagementCommanderImpl @Inject() (
...
+    val newEmailContacts = account.emailContacts.diff(emails.map(_.address)) // TODO(cam): email contacts aren't exposed (10/16/15), once they are we need to create an event for each email contact removed, or refactor AccountEventAction.AccountContactsChanged to take multiple contacts
+    val newUserContacts = if (account.userContacts.contains(userId)) {
+      val ownerIdOpt = if (account.userContacts.length == 1) Some(orgRepo.get(orgId).ownerId) else None
+      accountEventRepo.save(AccountEvent.simpleNonBillingEvent(
+        eventTime = clock.now,
+        accountId = orgId2AccountId(orgId),
+        attribution = attribution,
+        action = AccountEventAction.AccountContactsChanged(userAdded = ownerIdOpt, userRemoved = Some(userId), None, None)
+      ))
+      val newContacts = account.userContacts.diff(Seq(userId)) ++ ownerIdOpt.map(Seq[Id[User]](_)).getOrElse(Seq.empty[Id[User]])
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Are the explicit type signatures needed here?

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Also, `ownerIdOpt.toSeq` will do what you want here

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Is this code temporary? Once you hit the migration endpoint, the owner should always be in there, right? So there wouldn't be a need to add it in _ad hoc_ here.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: `ownerIdOpt.toSeq` will remove the need for types. They remove compile errors, I think type inference is weak when working with empty collections.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: you're right. owner enforcement wasn't spec'd before this code was written, and it makes sure if no one else, the owner is a contact. will improve.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Sweet. :+1:

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanManagementCommander.scala]: Looks like there was some confusion. https://app.asana.com/0/search/59244777063648/45938003030834 has `[pending cost] The owner of the org does not get the option to be removed from notifications`

**Pull request merged**